### DISCUSSION TITLE
About Add greptimedb to OpenTelemery Collector exporter

### DISCUSSION THREAD

**Discussion by JasonChen86899** [GreptimeTeam/greptimedb#1176] [Ideas]
Hello, greptimedb members.

I recently do something with opentelmetry collector and have an idea. Just make greptimedb to be an exporter of opentelemetry collector. opentelemetry data is all timeseries data inclulde trace metric log, so use greptimedb as backend store is a good idea and a good plan and try

**Comment by JasonChen86899** [comment]
If ok, I will go to opentelemetry community to make a issue or discussion

**Comment by killme2008** [comment]
Good idea. Looks like we have to implement the OpenTelemetry protocol in frontend

https://opentelemetry.io/docs/reference/specification/protocol/otlp/

We already support many protocols  https://docs.greptime.com/user-guide/protocols/overview
I don't know whether the OpenTelemetry collector supports other protocols.

**Reply by JasonChen86899** [reply]
Thank you for your reply, following is my detail thinking

1. openTelemetry  collector support exporters:  oltp jaeger prometheus stdout zipkin and have each adapter in client binary
2. openTelemetry collector can use grpc and **`make a transformation from oltp to greptimedb grpc protocol`** then send to greptimedb just like jaeger only add an adapter
3. I found greptimedb has already grpc-client `https://github.com/GreptimeTeam/greptimedb-client-go` that is good for doing something with otlp-go-client 🎉

**Reply by killme2008** [reply]
Thanks for your information. Let's do this work. It's kind full if you can create an issue in greptimedb and opentelemetry, then we can track the progress.

**Reply by JasonChen86899** [reply]
Great！Another question ,  I see only use **request.Metric** in `greptimedb-client-go`. How I can store tracing and log data with greptimedb . Is it just also with  **request.Metric** or need add models like tracing or log in client library?

**Reply by yuanbohan** [reply]
We currently only support metric-type data, but if you want to store log and trace data in greptimedb, you can temporarily use the metric model. You can put the data you need to query into  the `tag` column and set the information for display only in the `field` column. Then, you can use SQL to perform queries.
Client-go is still in the early development stage, and there may be some incompatible changes in the near future. Thank you for your continued interest in greptimedb. If you have any questions, please feel free to contact us here at any time.

**Reply by JasonChen86899** [reply]
Thx a lot！Looking forward to the new features of client-go！I hope I can participate in the design and discussion of oltp or tracking and log related functions in client-go. 

now two idea about this 
1. add tracing and log protocol in client-go
2. add another backend build on greptimedb just like [promscale](https://docs.timescale.com/promscale/latest/) which is another tsdb component.

**Reply by JasonChen86899** [reply]
Have Submitted a [issue](https://github.com/GreptimeTeam/greptimedb/issues/1288). Please review it
